WordPress/Classes/ViewRelated/Aztec/New Group/AztecVerificationPromptHelper.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+
+internal class AztecVerificationPromptHelper: NSObject {
+
+    private let accountService: AccountService
+    private let wpComAccount: WPAccount
+
+    private weak var displayedAlert: FancyAlertViewController?
+    private var completionBlock: AztecVerificationPromptCompletion?
+
+    typealias AztecVerificationPromptCompletion = (Bool) -> ()
+
+    init?(account: WPAccount?) {
+        guard let passedAccount = account,
+              let managedObjectContext = account?.managedObjectContext else {
+                return nil
+        }
+
+        self.accountService = AccountService(managedObjectContext: managedObjectContext)
+
+        guard accountService.isDefaultWordPressComAccount(passedAccount),
+              !passedAccount.emailVerified.boolValue else {
+                // if the post the user is trying to compose isn't on a WP.com account,
+                // or they're already verified, then the verification prompt is irrelevant.
+                return nil
+        }
+
+        self.wpComAccount = passedAccount
+
+        super.init()
+
+        NotificationCenter.default.addObserver(forName: .UIApplicationDidBecomeActive, object: nil, queue: nil) { [weak self] _ in
+            self?.updateVerificationStatus()
+        }
```

Comment:
@aerych @jklausa gentlemen, we've got an issue on this line. The NotificationCenter's blocks API looks like this

```
open func addObserver(forName name: NSNotification.Name?, object obj: Any?, queue: OperationQueue?, using block: @escaping (Notification) -> Swift.Void) -> NSObjectProtocol
```

In order to successfully unregister the observer, [we'd need to keep a reference to the object returned by this call, and use it later](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1411723-addobserverforname?language=objc).

Reply:
TIL! I'm surprised that the compiler doesn't yell about unused return value here 🤔